src/HotChocolate/ApolloFederation/src/ApolloFederation/Extensions/ProvidesDirectiveTypeDescriptorExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using HotChocolate.Language;
+using HotChocolate.Types;
+
+namespace HotChocolate.ApolloFederation.Extensions
+{
+    public static class ProvidesDirectiveTypeDescriptorExtensions
+    {
+        public static IObjectFieldDescriptor Provides(
+            this IObjectFieldDescriptor descriptor,
+            string fieldSet)
+        {
+            if (descriptor is null)
+            {
+                throw new ArgumentNullException(nameof(descriptor));
+            }
+
+            return descriptor.Directive(
+                WellKnownTypeNames.Provides,
+                new ArgumentNode(
+                    "fields",
+                    new FieldSetType().ParseResult(fieldSet)
```

Comment:
I dont think we should use the scalars like this. @michaelstaib is there already a abstract serializer for the base of this scalar?

Reply:
Well spotted @PascalSenn 🥇